itest/rust/src/rect2_test.rs

```diff
@@ -0,0 +1,102 @@
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+use std::fmt::Debug;
+
+use crate::itest;
+use godot::prelude::{inner::InnerRect2, *};
+
+#[itest]
+fn rect2_equiv_unary() {
+    let test_rects = [
+        Rect2::from_components(0.2, 0.3, 1.5, 0.9),
+        Rect2::from_components(0.2, 0.3, 1.5, 1.9),
+        Rect2::from_components(0.2, 0.3, 1.0, 1.9),
+        Rect2::from_components(4.2, 4.3, 1.5, 1.9),
+        Rect2::from_components(8.2, 8.3, 2.5, 2.9),
+        Rect2::from_components(8.2, 8.3, 2.5, 3.9),
+    ];
+    let test_vectors = [
+        Vector2::ZERO,
+        Vector2::new(0.0, 10.0),
+        Vector2::new(10.0, 0.0),
+        Vector2::new(10.0, 10.0),
+    ];
+    let test_reals = [0.0, 1.0, 10.0, 32.0];
+    let test_sides = [
+        RectSide::Left,
+        RectSide::Top,
+        RectSide::Right,
+        RectSide::Bottom,
+    ];
+
+    fn evaluate_mappings<T>(key: &str, a: T, b: T)
+    where
+        T: PartialEq + Debug,
+    {
+        assert_eq!(a, b, "{}: outer != inner ({:?} != {:?})", key, a, b);
+    }
+
+    for a in test_rects {
+        let inner_a = InnerRect2::from_outer(&a);
+
+        evaluate_mappings("abs", a.abs(), inner_a.abs());
+        evaluate_mappings("area", a.area() as f64, inner_a.get_area());
+        evaluate_mappings("center", a.center(), inner_a.get_center());
+        evaluate_mappings("has_area", a.has_area(), inner_a.has_area());
+
+        for b in test_rects {
+            evaluate_mappings("encloses", a.encloses(b), inner_a.encloses(b));
+            evaluate_mappings(
+                "intersects",
+                a.intersects(b, true),
+                inner_a.intersects(b, true),
+            );
+            // Check intersection without considering borders
+            evaluate_mappings(
+                "intersects",
+                a.intersects(b, false),
+                inner_a.intersects(b, false),
+            );
+            evaluate_mappings(
+                "intersection",
+                a.intersection(b).unwrap_or_default(),
+                inner_a.intersection(b),
+            );
+            evaluate_mappings("merge", a.merge(b), inner_a.merge(b));
+        }
+
+        for b in test_vectors {
+            evaluate_mappings("expand", a.expand(b), inner_a.expand(b));
+            evaluate_mappings("has_point", a.has_point(b), inner_a.has_point(b));
+        }
+
+        for b in test_reals {
+            evaluate_mappings("grow", a.grow(b as f32), inner_a.grow(b));
+
+            for c in test_reals {
+                for d in test_reals {
+                    for e in test_reals {
+                        evaluate_mappings(
+                            "grow_individual",
+                            a.grow_individual(b as f32, c as f32, d as f32, e as f32),
+                            inner_a.grow_individual(b, c, d, e),
+                        );
+                    }
+                }
+            }
```

Comment:
This runs 4^4 = 256 iterations. If someone adds values to `test_reals`, the time complexity explodes.

It's probably OK to have a 4-depth nested loop to test all combinations (as long as the number is bounded), but it might be more interesting to test some other values. For example include negative ones.

Maybe instead of `test_reals`, define a separate
```rs
let grow_values = [-1.0, 0.0, 1.0, 7.0];
```
that also ensures that people only change this when they actually want to affect the testing of `grow()`.

Reply:
Just getting precision errors like that:
```
ERROR: Panic msg:  assertion failed: `(left == right)`
  left: `Rect2 { position: Vector2 { x: 1.2, y: -0.7 }, size: Vector2 { x: -0.5, y: 1.9000001 } }`,
 right: `Rect2 { position: Vector2 { x: 1.2, y: -0.7 }, size: Vector2 { x: -0.5, y: 1.9 } }`: grow_individual: outer != inner (Rect2 { position: Vector2 { x: 1.2, y: -0.7 }, size: Vector2 { x: -0.5, y: 1.9000001 } } != Rect2 { position: Vector2 { x: 1.2, y: -0.7 }, size: Vector2 { x: -0.5, y: 1.9 } })
 ```
 How do you think we could fix this?

Reply:
@juliohq what do you think about my suggestion? 🙂 
It's one of the last things before this PR should be ready!

Reply:
@Bromeon I wonder if there's any way to replace the logic in `evaluate_mappings` with `is_equal_approx` or `assert_eq_approx!` while keeping the `inner` and `outer` parameters.

Reply:
You could use my suggestion from [here](https://github.com/godot-rust/gdext/pull/242#discussion_r1187879398), and call `assert_eq_approx!` inside; or am I misunderstanding you?

Reply:
I mean, the current `evaluate_mappings` uses `assert_eq!`, which takes a custom panic message and thus we can print the `outer` and `inner` variables, I'm not sure if `is_equal_approx` or `assert_eq_approx!` does the same.

Reply:
They do, even if it's not called "outer" and "inner" but "left" and "right". That's probably OK.

https://github.com/godot-rust/gdext/blob/65ce2c3730eddc867131c45e028c6e7d3e5afa95/godot-core/src/builtin/math.rs#L176-L194